### DISCUSSION TITLE
Zed on Chromebook

### DISCUSSION THREAD

**Discussion by codebird** [zed-industries/zed#14062] [Feature Requests]
I use zed on my desktop PC Ubuntu 22.04. Now traveling, I have a chromebook for travel, I can't get zed to work, it installs and everything is fine, but once I try to run it, it is a very small window, keeps on flickering whenever you hover it with the mouse and it can't be used. Low density makes the window bigger, but still not usable.

**Comment by codebird** [comment]
seems like it is a wayland issue. When I used wayland on my desktop ubuntu, zed stopped totally opening too, unless I force x11. Can't force x11 on chromebook :(

**Comment by davepar** [comment]
Same here. Installed quickly on ChromeOS 126, but starting up zed displays a tiny window (1/6 size?), that's completely unusable. And it's continuously alternating between a new project wizard and "open a project" screen.

Side note: I'm excited to try Zed. VSCode is an enormous memory hog, which isn't great for a Chromebook. I've been using Sublime Text instead, but it doesn't seem as snappy as it used to be.

**Comment by codebird** [comment]
I went to `chrome://flags `enabled gpu for crostini JIC, to no avail :( I'll keep on tinkering...

**Comment by 0skxr** [comment]
I actually have the same problem.
Funnily enough it works wwhen launching it from weston.
![image](https://github.com/user-attachments/assets/d22f359e-6a69-4bc0-9728-6b9c2a0f4075)

**Comment by davepar** [comment]
I found this, but haven't tried it yet: https://chromeos.dev/en/linux/linux-on-chromeos-faq#why-are-windows-sometimes-tinyfuzzy

**Comment by codebird** [comment]
Now I am getting a pop up on launch, saying that the gpu is an emulated software gpu and it is unsupported :(

**Comment by 0skxr** [comment]
I think it might be an issue with the compositor rather then the gpu, since it works in weston.

**Comment by calionauta** [comment]
Same here.

**Reply by calionauta** [reply]
Hi there, any news? 🙏

**Comment by davepar** [comment]
There is more info in this thread/comment on setting the display DPI and installing Intel GPU drivers: https://github.com/zed-industries/zed/issues/13847#issuecomment-2304571096
However, I tried setting the DPI and I still get a tiny window.

**Comment by LaPingvino** [comment]
How I manage to use it on ChromeOS:
1. Turn on GPU support for Crostini
2. Install the Virtio GPU drivers and Vulkan
3. Run Zed with WAYLAND_DISPLAY set to empty (e.g. WAYLAND_DISPLAY= zeditor) to turn off Wayland

Hope I didn't forget anything there

I hope a way can be found to make it work better by default...

**Reply by mfouesneau** [reply]
This does not work for me.

**Reply by LaPingvino** [reply]
I got it to work with Arch on Crostini that way. Now I am back on Debian, and using Nix, and don't manage to get it running again... will keep you up to date about what I discover while trying to fix this! In short, it looks like a virtio/vulkan/wayland issue.

**Comment by mfouesneau** [comment]
Any new information?

**Reply by LaPingvino** [reply]
Can you post the output of `lspci -nn` and `vulkaninfo` here to help the Zed devs etc understand what's going on?

**Comment by LaPingvino** [comment]
[lspci.txt](https://github.com/user-attachments/files/20645366/lspci.txt)
[vulkaninfo.txt](https://github.com/user-attachments/files/20645367/vulkaninfo.txt)

Let me put mine here to give some context. I have managed to get virtio GPU to show up here before, but it looks like Zed struggles with some parts of this driver support. The default situation seems to be completely unsupported.

**Comment by mfouesneau** [comment]
[lspci-nn.txt](https://github.com/user-attachments/files/20780611/lspci-nn.txt)
[vulkaninfo.txt](https://github.com/user-attachments/files/20780612/vulkaninfo.txt)

**Reply by LaPingvino** [reply]
yeah the virtio GPU shows up, so I think with that vulkaninfo the devs should be able to do something.

**Reply by mfouesneau** [reply]
Thanks. I hope that's an easy fix.

**Comment by matthyx** [comment]
hello, I'm interested as well

**Reply by LaPingvino** [reply]
So, it looks like you need a set of conditions to get it to work: GPU support on, Vulkan on, Vulkan support in OS, Wayland off. When using Nix, you want the nixVulkan helper tool, this might also work if you don't use anything else from Nix.

**Reply by mfouesneau** [reply]
GPU support is already on in the flags. Not using nix.

**Reply by LaPingvino** [reply]
You can force Wayland off with WAYLAND_DISPLAY= (setting it to empty). Debian doesn't seem to have the necessary virtio gpu driver, which nixVulkan neatly substitutes.

**Reply by mfouesneau** [reply]
The code hangs without UI showing up.  (WAYLAND_DISPLAY and WAYLAND_DISPLAY_LOW_DENSITY unset)

**Reply by LaPingvino** [reply]
Can you check if you have a virtio gpu driver installed? the nixVulkanIntel command works as a possible replacement. https://gist.github.com/spundun/f01346a8d8789be861597690c7a849b8 this contains instructions otherwise, didn't test those.

**Reply by mfouesneau** [reply]
Fresh container, checked vulkan info, and running vkcube.
If I do not remove WAYLAND_DISPLAY I get a window with unsupported GPU
If I do, I get nothing.

```
2025-09-09T19:47:46+02:00 INFO  [zed] ========== starting zed version 0.202.7, sha cd9b0b5 ==========
2025-09-09T19:47:47+02:00 INFO  [zed] Using git binary path: None
2025-09-09T19:47:47+02:00 INFO  [crashes] spawning crash handler process
2025-09-09T19:47:47+02:00 INFO  [extension_host] extensions updated. loading 1, reloading 0, unloading 0
2025-09-09T19:47:47+02:00 INFO  [gpui::platform::linux::platform] activate is not implemented on Linux, ignoring the call
2025-09-09T19:47:47+02:00 INFO  [auto_update] auto-update check failed: error:Aborting. Could not find rsync which is required for auto-updates.
2025-09-09T19:47:47+02:00 ERROR [client] DBus error service error org.freedesktop.zbus.Error: The name org.freedesktop.secrets was not provided by any .service files
2025-09-09T19:47:47+02:00 INFO  [crashes] connected to crash handler process after 100ms
2025-09-09T19:47:47+02:00 INFO  [crashes] crash handler registered
2025-09-09T19:47:47+02:00 INFO  [gpui::platform::blade::blade_renderer] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PreMultiplied }
2025-09-09T19:47:47+02:00 INFO  [zed::zed] Using GPU: GpuSpecs { is_software_emulated: true, device_name: "llvmpipe (LLVM 15.0.6, 256 bits)", driver_name: "llvmpipe", driver_info: "Mesa 22.3.6 (LLVM 15.0.6)" }
2025-09-09T19:47:53+02:00 ERROR [agent2::agent] Failed to authenticate provider: Amazon Bedrock: DBus error service error org.freedesktop.zbus.Error: The name org.freedesktop.secrets was not provided by any .service files
2025-09-09T19:47:53+02:00 ERROR [agent_ui] No models available
2025-09-09T19:47:53+02:00 ERROR [agent_ui] No models available
2025-09-09T19:47:53+02:00 ERROR [agent2::agent] Failed to authenticate provider: Anthropic: DBus error service error org.freedesktop.zbus.Error: The name org.freedesktop.secrets was not provided by any .service files
2025-09-09T19:47:53+02:00 ERROR [agent2::agent] Failed to authenticate provider: DeepSeek: DBus error service error org.freedesktop.zbus.Error: The name org.freedesktop.secrets was not provided by any .service files
2025-09-09T19:47:53+02:00 ERROR [agent2::agent] Failed to authenticate provider: Google AI: DBus error service error org.freedesktop.zbus.Error: The name org.freedesktop.secrets was not provided by any .service files
2025-09-09T19:47:53+02:00 ERROR [agent2::agent] Failed to authenticate provider: Mistral: DBus error service error org.freedesktop.zbus.Error: The name org.freedesktop.secrets was not provided by any .service files
2025-09-09T19:47:53+02:00 ERROR [agent2::agent] Failed to authenticate provider: OpenAI: DBus error service error org.freedesktop.zbus.Error: The name org.freedesktop.secrets was not provided by any .service files
2025-09-09T19:47:53+02:00 ERROR [agent2::agent] Failed to authenticate provider: OpenRouter: DBus error service error org.freedesktop.zbus.Error: The name org.freedesktop.secrets was not provided by any .service files
2025-09-09T19:47:53+02:00 ERROR [agent2::agent] Failed to authenticate provider: Vercel: DBus error service error org.freedesktop.zbus.Error: The name org.freedesktop.secrets was not provided by any .service files
```

**Reply by mfouesneau** [reply]
```
sudo lshw -c video
  *-display:0               
       description: Display controller
       product: Red Hat, Inc.
       vendor: Red Hat, Inc.
       physical id: 1
       bus info: pci@0000:00:01.0
       version: 01
       width: 64 bits
       clock: 33MHz
       capabilities: bus_master cap_list
       configuration: driver=virtio-pci latency=0
       resources: iomemory:20-1f irq:9 memory:d0800000-d0807fff memory:200000000-2ffffffff
  *-display:1
       description: Display controller
       product: Virtio 1.0 GPU
       vendor: Red Hat, Inc.
       physical id: 2
       bus info: pci@0000:00:02.0
       version: 01
       width: 64 bits
       clock: 33MHz
       capabilities: bus_master cap_list
       configuration: driver=virtio-pci latency=0
       resources: iomemory:40-3f irq:10 memory:d0808000-d080ffff memory:400000000-5ffffffff
       ```

**Reply by mfouesneau** [reply]
Debian versions of Crostini are now too old to build the drivers.

**Reply by LaPingvino** [reply]
So I guess using the nixVulkanIntel command might be the only working option

**Reply by mfouesneau** [reply]
Do you have a procedure?

**Comment by matthyx** [comment]
Success! I am able to run Zed 0.205.8 on my Chromebook, here is how I did it:

Add backports to apt sources: `sudo apt edit-sources`, after adding the last line it looks like:
```
# Generated by distrobuilder
deb https://deb.debian.org/debian bookworm main
deb https://deb.debian.org/debian bookworm-updates main
deb https://deb.debian.org/debian-security/ bookworm-security main
deb http://deb.debian.org/debian bookworm-backports main contrib non-free non-free-firmware
```

Then run `sudo apt update` and finally `sudo apt install mesa-vulkan-drivers/bookworm-backports`.

I've restarted `penguin` after enabling the flag `[#crostini-gpu-support](chrome://flags/#crostini-gpu-support)` and now Zed works !
<img width="1919" height="1031" alt="image" src="https://github.com/user-attachments/assets/db215fc0-66e1-409b-8f41-a56c52a6ef26" />

**Reply by nooobss** [reply]
sorry that tips doesnt work on my chromebook with trixie debian version. i have changed keyword "bookworm" to "trixie"

**Comment by mfouesneau** [comment]
@codebird thanks! This solves the GPU part. 
However, my app opens in a tiny window that I cannot really use... If I move it between physical screens, it eventually becomes responsive and resizable. It's probably a WAYLAND issue, but if I unset the WAYLAND_DISPLAY variable, it crashes.
If I move the app to a different screen again, it reduces and becomes unusable. It's like it has a hard time understanding its own size.

**Reply by rsamborski** [reply]
@mfouesneau I was able to get Zed running in a normal window, but there is still an issue with using the mouse. See here for more details: #37878

**Reply by mfouesneau** [reply]
My mouse nominally works (I think). When I move the window, I can see a shade that is beyond (right, bottom). Maybe there is something with the density/resolution? (I cannot change the density to low with Zed apparently)

**Reply by Zorro92** [reply]
> @codebird thanks! This solves the GPU part. However, my app opens in a tiny window that I cannot really use... If I move it between physical screens, it eventually becomes responsive and resizable. It's probably a WAYLAND issue, but if I unset the WAYLAND_DISPLAY variable, it crashes. If I move the app to a different screen again, it reduces and becomes unusable. It's like it has a hard time understanding its own size.

Same. That fix of unsettling WAYLAND_DISPLAY USED to work. But it no longer does 😥 I'm assuming a recent ChromeOS update broke this workaround.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Reply by studyingegret** [reply]
Isn't this in the [Feature Requests](https://github.com/zed-industries/zed/discussions/categories/feature-requests) category?

**Comment by johnozbay** [comment]
Hello all 👋🏻

I cannot thank you enough for all the helpful information posted on this thread.

Thanks to [this Zed update](https://github.com/zed-industries/zed/pull/46758) I managed to get Zed to work even on my ARM Chromebook, and wrote a little tutorial / guide blog post about it to save everyone some time : 

https://blog.johnozbay.com/running-zed-editor-on-arm-chromebooks-mali-g925-crostini-2.html

Let me know if you run into any issues, and I'd be happy to update the blog post as well.

Hoping this helps!

**Reply by Zorro92** [reply]
Very nice write up John. I wish I still ran ChromeOS on my Chromebook to try this. Unfortunately I've moved on and have since unlocked my device and installed a flavor of atomic fedora on my Chromebook (running very nice btw). I do hope one day ChromeOS gets official support for Zed.

**Reply by shibumi** [reply]
Hi,

I tried to follow this guide, but run into this issue here:

```
❯ sudo apt install -t bookworm-backports \
  mesa-vulkan-drivers \
  mesa-utils \
  libgl1-mesa-dri \
  libegl-mesa0 \
  libglx-mesa0 \
  mesa-libgallium
mesa-vulkan-drivers is already the newest version (25.0.7-2).
mesa-vulkan-drivers set to manually installed.
Solving dependencies... Error!
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

Unsatisfied dependencies:
 mesa-libgallium : Depends: libllvm15 but it is not installable
Error: Unable to correct problems, you have held broken packages.
Error: The following information from --solver 3.0 may provide additional context:
   Unable to satisfy dependencies. Reached two conflicting decisions:
   1. mesa-libgallium:amd64=25.0.7-2~bpo12+1 is selected for install
   2. mesa-libgallium:amd64=25.0.7-2~bpo12+1 Depends libllvm15
      but none of the choices are installable:
      [no choices]
 ```
      
Looks like this no longer works :/ Also, I am running container less Crostini (the new native VM).

**Reply by johnozbay** [reply]
Hey @shibumi 👋🏻

I'm still quite new to all this too, so forgive me for not being able to offer lots of help given the number of variables that could be causing this. One thing I noticed tho, are you on amd64 or arm? Asking because of this line : 
> mesa-libgallium:amd64=25.0.7-2~bpo12+1 is selected for install
 
Because this blog post is specifically about arm architecture. (and my linux-fu isn't super good, so I'm trying to figure out with the limited info I have, forgive me hah) It seems like the mesa support is super finicky right now, and depends even on the mood of the laptop. So it could even be a Baguette vs Crostini thing.

**Reply by johnozbay** [reply]
@shibumi can confirm it continues to work as of today at least on my stock chromeos crostini system, and just today updated Zed to the latest v1.0.1 and [updated the blog post with instructions](https://blog.johnozbay.com/running-zed-editor-on-arm-chromebooks-mali-g925-crostini-2.html#updating-zed--enabling-auto-updates) for others to be able to update to the new version easily.

**Comment by mfouesneau** [comment]
@johnozbay thanks. This works. I use `Xft.dpi:` 144` 

I found that closing zed, I need to run the command twice to get a new editor.

**Comment by NikZzzonked** [comment]
On my Chromebook on Intel I needed only to enable GPU for Crostini in `chrome://flags` and run `MESA_VK_WSI_DEBUG=sw WAYLAND_DISPLAY='' zed`

**Reply by shibumi** [reply]
Did you try this with Containerless Crostini aka Baguette?

**Reply by NikZzzonked** [reply]
I use the default Penguin.
